src/createAnimatedComponent.tsx

```diff
@@ -402,25 +404,28 @@ export default function createAnimatedComponent(
     _updateFromNative(props: StyleProps) {
       if (options?.setNativeProps) {
         // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        options.setNativeProps(this._component!, props);
+        options.setNativeProps(this._component as ComponentRef, props);
       } else {
         // eslint-disable-next-line no-unused-expressions
-        this._component?.setNativeProps?.(props);
+        (this._component as ComponentRef)?.setNativeProps?.(props);
       }
     }
 
     _getViewInfo() {
-      let viewTag: number | null;
+      let viewTag: number | HTMLElement | null;
       let viewName: string | null;
       let shadowNodeWrapper: ShadowNodeWrapper | null = null;
       let viewConfig;
       // Component can specify ref which should be animated when animated version of the component is created.
       // Otherwise, we animate the component itself.
-      const component = this._component?.getAnimatableRef
-        ? this._component.getAnimatableRef()
-        : this;
-      if (isWeb()) {
-        viewTag = findNodeHandle(component);
+      const component =
+        'getAnimatableRef' in this._component!
+          ? this._component.getAnimatableRef!()
```

Comment:
Is it possible to somehow convert this snippet of code so that we don't have to use `!`? This is meant for "unsafe" operations when we are sure that the property exists (e.g. JSI bindings injected from C++ side). Can we use TypeScript to ensure that this property exists (either with ternary conditional or using `?`)?